src/components/entity/ha-entity-state-content-picker.ts

```diff
@@ -111,63 +108,125 @@ export class HaStateContentPicker extends LitElement {
 
   @property() public helper?: string;
 
-  @query(".container", true) private _container?: HTMLDivElement;
-
-  @query("ha-combo-box", true) private _comboBox!: HaComboBox;
-
-  @state() private _opened = false;
+  @query("ha-generic-picker", true) private _picker?: HaGenericPicker;
 
   private _editIndex?: number;
 
-  private _options = memoizeOne(
+  private _getItems = memoizeOne(
     (entityId?: string, stateObj?: HassEntity, allowName?: boolean) => {
       const domain = entityId ? computeDomain(entityId) : undefined;
-      return [
+      const items: PickerComboBoxItem[] = [
         {
+          id: "state",
           primary: this.hass.localize(
             "ui.components.state-content-picker.state"
           ),
-          value: "state",
+          search_labels: {
+            primary: this.hass.localize(
+              "ui.components.state-content-picker.state"
+            ),
+            id: "state",
```

Comment:
This is not needed. `search_labels` are there to search for stuff thats not in the object. But primary and id is already there. default search keys will manage it well.